rpcs3/Emu/RSX/Common/ProgramStateCache.h

```diff
@@ -348,33 +343,41 @@ class program_state_cache
 		// NOTE: Linking is much slower than decompilation step, so always decompile at least 1 unit
 		// TODO: Use try_lock instead
 		bool busy = false;
-		{
-			u32 count = 0;
-			std::lock_guard lock(m_decompiler_mutex);
+		u32 count = 0;
```

Comment:
This isn't the same decompiler routine used to decompile shaders when loading so I don't think modifying this function was necessary. Wouldn't hurt anyway I suppose, but this then raises another issue: there are several cases of mutex logic to access a queue container, both here and in the decompile routine. Rpcs3 already has lock free containers you can use for this, removing the need for all the synchronization primitives.

Reply:
I only messed with it because I noticed it was synchronizing before calling search_vertex_program() and search_fragment_program(), which shouldn't be needed anymore (at least on Vulkan, that is what I'm worried about with OpenGL)

I can drop these changes if you think its better.

Reply:
Drop it for simplicity, unless you intend to rewrite it to be lock free, then its fine.

Reply:
Gave a shot at making it lock-free.